src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -131,14 +132,50 @@ public NamedPipeTest_CurrentUserOnly_Windows(TestAccountImpersonator testAccount
             _testAccountImpersonator = testAccountImpersonator;
         }
 
+        [OuterLoop]
+        [ConditionalFact(nameof(IsAdminOnSupportedWindowsVersions))]
+        public async Task Connection_UnderDifferentUsers_CurrentUserOnlyOnServer_InvalidClientConnectionAttempts_DoNotBlockSuccessfulClient()
+        {
+            string name = PipeStreamConformanceTests.GetUniquePipeName();
+            ManualResetEvent @event = new ManualResetEvent(false);
+
+            Task invalidClient = Task.Run(() =>
+            {
+                // invalid non-current user tries to connect to server.
+                _testAccountImpersonator.RunImpersonated(() =>
+                {
+                    using var client = new NamedPipeClientStream(".", name, PipeDirection.In, PipeOptions.Asynchronous);
+                    @event.WaitOne();
+                    Assert.Throws<UnauthorizedAccessException>(() => client.Connect());
```

Comment:
with only one task here, I wouldn't be surprised if eg., 99.9999% of the time, one consistently runs before the other (eg., this has to go through Assert.Throws) which isn't giving you the concurrency you're looking for.

Reply:
Would this suffice? 
```suggestion
                    Assert.Throws<UnauthorizedAccessException>(() =>
                    {
                        @event.WaitOne();
                        client.Connect();
                    });
```

Reply:
Probably good enough!